packages/loot-core/src/server/budget/goaltemplates.ts

```diff
@@ -231,14 +230,20 @@ async function processTemplate(month, force, category_templates) {
     //if overwrite is not preferred, set cell to original value
     for (let l = 0; l < originalCategoryBalance.length; l++) {
       await setBudget({
-        category: originalCategoryBalance[l].cat.id,
+        category: originalCategoryBalance[l].category,
         month,
         amount: originalCategoryBalance[l].amount,
       });
       //if overwrite is not preferred, remove template errors for category
       let j = errors.length;
       for (let k = 0; k < j; k++) {
-        if (errors[k].includes(originalCategoryBalance[l].cat.name)) {
+        if (
```

Comment:
In the case overwrite is preferred, this filters out categories that don't have templates and income categories and applies the originally budgeted amount.